pkgs/top-level/all-packages.nix

```diff
@@ -9761,6 +9761,10 @@ in
 
   leaps = callPackage ../development/tools/leaps { };
 
+  ### DEVELOPMENT / JAVA MODULES
+
+  javaPackages = recurseIntoAttrs (callPackage ./java-packages.nix { });
```

Comment:
This is pretty much the bare minimum to build anything with Maven :) If we can deal with `callPackage` instead of `callPackages`, let's merge...